### ISSUE TITLE
Add reduction backprop methods

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/libnd4j#718]
Currently, as far as I can tell, reductions (mean, sum, stdev, etc) are using legacy op execution, and we don't have any native backprop methods... Instead, for backprop here we are implementing it manually:

https://github.com/deeplearning4j/nd4j/blob/79eb921012122cbe903aecb609cf9e960234aa3b/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/accum/Variance.java#L150-L160

**Comment by raver119** [comment]
Yep, at this moment we're silently using legacy ops everywhere we need those.
